entity-data-service-rx-client/src/main/java/org/hypertrace/entity/data/service/rxclient/EntityDataCachingClient.java

```diff
@@ -112,9 +121,14 @@
     private UpsertCondition condition;
 
     private void executeUpdate() {
-
-      EntityDataCachingClient.this.pendingEntityUpdates.remove(entityKey);
-
+      // Acquire write lock to ensure no more modification of this update
+      Lock lock = pendingUpdateStripedLock.get(entityKey).writeLock();
+      try {
+        lock.lock();
+        EntityDataCachingClient.this.pendingEntityUpdates.remove(entityKey);
+      } finally {
+        lock.unlock();
+      }
       Single<Entity> updateResult =
           EntityDataCachingClient.this.createOrUpdateEntity(entityKey, condition).cache();
       EntityDataCachingClient.this.cache.put(entityKey, updateResult);
```

Comment:
The above method at L133 `EntityDataCachingClient.this.createOrUpdateEntity(entityKey, condition).cache();` internally already adds the updated result into the cache:
```
  private Single<Entity> createOrUpdateEntity(
      EntityKey entityKey, UpsertCondition upsertCondition) {
    Single<Entity> updateResult =
        this.upsertEntityWithoutCaching(entityKey, upsertCondition).cache();
    EntityDataCachingClient.this.cache.put(entityKey, updateResult);
    return updateResult;
  }
```
Hence L134 is redundant.